packages/backend/src/graphql/resolvers/incoming_payment.test.ts

```diff
@@ -288,7 +304,125 @@ describe('Incoming Payment Resolver', (): void => {
       }
       expect(createSpy).toHaveBeenCalledWith({
         ...input,
-        tenantId
+        tenantId,
+        initiationReason: IncomingPaymentInitiationReason.Admin
+      })
+    })
+
+    describe('tenant boundaries', (): void => {
+      test('operator can label incoming payment as card payment', async (): Promise<void> => {
+        const payment = await createIncomingPayment(deps, {
+          walletAddressId,
+          client,
+          tenantId,
+          initiationReason: IncomingPaymentInitiationReason.Admin
+        })
+
+        const createSpy = jest
+          .spyOn(incomingPaymentService, 'create')
+          .mockResolvedValueOnce(payment)
+
+        const input = {
+          walletAddressId,
+          isCardPayment: true
+        }
+
+        const query = await appContainer.apolloClient
+          .query({
+            query: gql`
+              mutation CreateIncomingPayment(
+                $input: CreateIncomingPaymentInput!
+              ) {
+                createIncomingPayment(input: $input) {
+                  payment {
+                    id
+                    walletAddressId
+                  }
+                }
+              }
+            `,
+            variables: { input }
+          })
+          .then(
+            (query): IncomingPaymentResponse =>
+              query.data?.createIncomingPayment
+          )
+
+        expect(createSpy).toHaveBeenCalledWith({
+          walletAddressId: input.walletAddressId,
+          tenantId,
+          initiationReason: IncomingPaymentInitiationReason.Card
+        })
+        expect(query).toEqual({
+          __typename: 'IncomingPaymentResponse',
+          payment: {
+            __typename: 'IncomingPayment',
+            id: payment.id,
+            walletAddressId
+          }
+        })
+      })
+
+      test('tenant cannot label incoming payment as card payment', async (): Promise<void> => {
+        const tenant = await createTenant(deps)
+        const tenantWalletAddress = await createWalletAddress(deps, {
+          tenantId: tenant.id
+        })
+        const payment = await createIncomingPayment(deps, {
+          walletAddressId: tenantWalletAddress.id,
+          client,
+          tenantId: tenant.id,
+          initiationReason: IncomingPaymentInitiationReason.Admin
+        })
```

Comment:
Do we need to create this incoming payment here, before calling the mutation?
(same goes for the test above)

Reply:
Frankly, I copied this from the pattern of other tests in this file but it's not really necessary.